### DISCUSSION TITLE
Editing road restrictions

### DISCUSSION THREAD

**Discussion by paulens12** [osmandapp/OsmAnd#11891] [Ideas]
I use OsmAnd for navigation and there's one huge issue with it: it often doesn't know that I can't turn left on an intersection, so it keeps leading me the wrong way and then I have to stop and study the map by myself to find the right way. I'd like to be able to add such restrictions manually - it probably won't save the trip I'm currently taking, but it could help someone else who ends up in the same spot. The thing with desktop editing is you have to take note, then get back home, find the spot on the map and edit it - and I usually forget to do that. Another potential advantage would be instant feedback, where you add the "no turning left here" restriction (just as an example) and the navigation automatically updates with the local change, before it's even pushed to OSM. Then, instead of stopping to study the map, I'd stop to make the tiny correction and then continue using the navigation.

**Comment by pebogufi** [comment]
I use favourites to make private notes.

**Reply by paulens12** [reply]
yeah but private notes will not affect the navigation in any way, it will keep suggesting you go the wrong way. They also won't help other drivers using OsmAnd or other OSM-based apps. What I'm talking about is not just a note on the screen, I'm talking about intrinsic information. Each road segment is a "path" in OSM terms, and each intersection is a "point". Points can have restrictions like this as part of their metadata, and the navigation algorithm takes that into account so it will never suggest you turn left in an intersection if the point's metadata says you can't.

**Comment by vshcherb** [comment]
Hm... I had similar situation but there was no choice just to edit OSM and use osmand live to update maps later...

**Comment by paulens12** [comment]
yeah, that's why I'm suggesting this as a new feature :) I just keep forgetting to update the map when I get home, it would be so much better if I could do it right there on the road, not only because I wouldn't forget to do it, but also because I would be on the spot and could verify details easily.
________________________________
